cluster-autoscaler/utils/drain/drain.go

```diff
@@ -33,6 +33,12 @@ const (
 	PodDeletionTimeout = 12 * time.Minute
 )
 
+const (
+	// DeleteNonReplicatedPodEnabledKey - if pod has this annotation, the node it is in can be drained even
+	// if this pod is not replicated.
+	DeleteNonReplicatedPodEnabledKey = "cluster-autoscaler.kubernetes.io/delete-non-replicated-pod-enabled"
```

Comment:
This overrides a single specific check. Maybe we should instead go for something like `cluster-autoscaler.kubernetes.io/safe-to-evict` which would also pass the kube-system and local storage checks in the same function? Or at least the kube-system one, because current solution won't work for pods in kube-system namespace.

WDYT @Henrod @felipejfc @mwielgus ?

Reply:
I agree @MaciekPytel